### PR TITLE
Adding BoBot to run alongside Enchant

### PR DESCRIPTION
**Pull request by theBGuy**

### CODE REVIEW

File: d2bs/kolbot/libs/common/Config.js [hunk 1]
```diff
@@ -185,7 +185,6 @@ var Config = {
 	MakeRoom: true,
 	ClearInvOnStart: true,
 	FastPick: false,
-	ManualPlayPick: false,
```
Comment (noah-) [inline d2bs/kolbot/libs/common/Config.js:188]: Is this line removed by mistake?

Comment (casualBotter) [reply d2bs/kolbot/libs/common/Config.js:188]: Yeah that was a mistake.

Comment (casualBotter) [reply d2bs/kolbot/libs/common/Config.js]: fixed it

File: d2bs/kolbot/libs/bots/BoBot.js [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+		}
+
+		var partyUnit,
+			unit = getUnit(0, nick);
+
+		if (getDistance(me, unit) > 10) {
+			say("Get closer.");
+
+			return false;
+		}
```
Comment (5noop) [inline d2bs/kolbot/libs/bots/BoBot.js:27]: Isn't the unit supposed to be waiting at a wp and not in town ?

File: d2bs/kolbot/libs/bots/BoBot.js [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+
+			// wait until party area is readable?
+
+			if ([1,40, 75, 103, 109].indexOf(partyUnit.area) > -1) {
+				say("Cannot Bo in town.");
+				print("cannot bo in town");
+				return false
+			} else{
+				print(partyUnit.area); //for debug
+				Pather.useWaypoint(partyUnit.area);
```
Comment (5noop) [inline d2bs/kolbot/libs/bots/BoBot.js:40]: Should check if `partyUnit.area` has actually a waypoint. If not you could handle the case by telling the player he should get close to a waypoint.

File: d2bs/kolbot/libs/bots/BoBot.js [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+
+		unit = getUnit(1);
+
+		if (unit) {
+			do {
+				if (unit.getParent() && unit.getParent().name === nick) { // merc or any other owned unit
+					Precast.doPrecast(true);
+				}
+			} while (unit.getNext());
+		}
```
Comment (5noop) [inline d2bs/kolbot/libs/bots/BoBot.js:68]: Is this part really needed ? Merc and summons should have bo once you cast it on the player. Imagine if a player has 10 summons the barb will end up casting bo/bc/shout 11 times in a row.

File: d2bs/kolbot/libs/bots/BoBot.js (hunk 2 above)
Comment (casualBotter) [reply d2bs/kolbot/libs/bots/BoBot.js:27]: It waits in town until called, I am having trouble if someone calls bo and they aren't right on the wp it says they are too far. Any ideas on how to fix that i've been trying to alter the range for getunit but its not helping

File: d2bs/kolbot/libs/bots/BoBot.js (hunk 3 above)
Comment (casualBotter) [reply d2bs/kolbot/libs/bots/BoBot.js:40]: Yeah you're right I added a check for that and will update when I figure out distance issue

File: d2bs/kolbot/libs/bots/BoBot.js (hunk 4 above)
Comment (casualBotter) [reply d2bs/kolbot/libs/bots/BoBot.js:68]: Also right i got rid of this.

File: d2bs/kolbot/libs/bots/BoBot.js (hunk 2 above)
Comment (5noop) [reply d2bs/kolbot/libs/bots/BoBot.js:27]: The unit is the player, it should probably be waiting for you near a waypoint so you won't be able to use `getUnit()` which has a limited range.

You could remove this part, keep `getParty()` and make sure the player's area has a waypoint and is not a town.

File: d2bs/kolbot/libs/bots/BoBot.js [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+		if (!Misc.inMyParty(nick)) {
+			say("Accept party invite, noob.");
+
+			return false;
+		}
+
+		var partyUnit,
+			unit = getUnit(0, nick);
+
+		if (!unit) {
```
Comment (5noop) [inline d2bs/kolbot/libs/bots/BoBot.js:23]: Don't use `getUnit()` here, if it succeeds we jump to line 46 and barb will precast in town.

File: d2bs/kolbot/libs/bots/BoBot.js [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+
+			// wait until party area is readable?
+
+			if ([1,40, 75, 103, 109].indexOf(partyUnit.area) > -1) {
+				say("Cannot Bo in town.");
+				
+				return false
+			} else{
+				print(partyUnit.area); //for debug
+				if([3, 4, 5, 6, 27, 29, 32, 35, 48, 42, 57, 43,
```
Comment (5noop) [inline d2bs/kolbot/libs/bots/BoBot.js:34]: This can be an `if else ()`

File: d2bs/kolbot/libs/bots/BoBot.js [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+
+				if (shitList.indexOf(command[1]) > -1) {
+					say("No " + command[0] + " for the shitlisted.");
+
+					break;
+				}
+
+				say("Commands" +
+						(Config.BoBot.Trigger[0] ? " | Bo: " + Config.BoBot.Trigger[0] : "") + 
+						"You must go to a Waypoint and stand on the square.");
```
Comment (5noop) [inline d2bs/kolbot/libs/bots/BoBot.js:193]: You could add a space or | at the beginning for readability

File: d2bs/kolbot/libs/bots/BoBot.js [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+						say("You went too far away.");
+
+						return false;
+					}
+					Precast.doPrecast(true);
+				}
+			} while (unit.getNext());
+		} else {
+			say("Couldn't find you, champ.");
+		}
```
Comment (5noop) [inline d2bs/kolbot/libs/bots/BoBot.js:59]: This part could be moved after line 38

File: d2bs/kolbot/libs/bots/BoBot.js (hunk 5 above)
Comment (casualBotter) [reply d2bs/kolbot/libs/bots/BoBot.js:23]: What do I use? I tried taking out that part but I got errors

**Comment by casualBotter** [comment]
> You could run a linter (eslint and jslint files are in the readme) to fix the indentation etc, otherwise it looks good to me 👍

I wasn't sure how to do the eslint or jslint but I went through and fixed the indentation mistakes I saw.

**Comment by noah-** [comment]
if you use notepad++ you can get eslint or jslint plugin and use the preset in the repo readme for the settings

**Pull request closed without merging**